docs/concepts/advanced/rendering_js_css.md

```diff
@@ -220,3 +220,201 @@ with [`Component.render()`](#TODO) and inserting them into larger structures.
     - Directly passing rendered HTML to [`render_dependencies()`](#TODO)
 3. If you pre-render one component to pass it into another, the pre-rendered component must be rendered with
    [`render_dependencies=False`](#TODO).
+
+## Render types
```

Comment:
Moved this here from `rendering_components.md`, and instead in `rendering_components.md` there is only a short overview of what each mode does.

<img width="1331" alt="Screenshot 2025-04-28 at 21 47 23" src="https://github.com/user-attachments/assets/3f9cdc6c-b461-4a30-8232-971935d72316" />